app/Gio.java

```diff
@@ -65,4 +66,17 @@ static void wakeupMainThread() {
 	}
 
 	static private native void scheduleMainFuncs();
+
+	static Intent startForegroundService(Context ctx, String activityClassName, String title, String text) {
+		Intent intent = new Intent();
+		try {
+			intent.setClass(ctx, ctx.getClassLoader().loadClass(activityClassName));
```

Comment:
You named the parameter activityClassName (as I'd expect), but use it as the service class name. See also comment below about GioActivity.